src/fides/api/models/datasetconfig.py

```diff
@@ -55,93 +55,71 @@ class DatasetConfig(Base):
     )
 
     @classmethod
-    def upsert_with_ctl_dataset(
-        cls, db: Session, *, data: Dict[str, Any]
-    ) -> "DatasetConfig":
+    def create_or_update(cls, db: Session, *, data: Dict[str, Any]) -> "DatasetConfig":  # type: ignore[override]
```

Comment:
I consolidated the logic of `create_or_update` and `upsert_with_ctl_dataset` into one. When working with `DatasetConfigs` you're always talking about the related `ctl_dataset` so it didn't make sense to keep this separate.